src/Logger/index.ts

```diff
@@ -21,21 +21,37 @@ export type Serializer = (msg: any) => string
 // Stamper allows extention of the log event, you can sumply any number of stampers which will all apply their stamp to a message
 export type Stamper = () => any
 
+// Options contain all the required parts of the Logger
+export type Options = {
+    level: LogLevel
+    serialize: Serializer
+    stamps: Stamper[]
+    write: Writer
+}
+
+// Parameters provides a way to instantiate a Logger with only some of the required parts
+export interface Parameters extends Partial<Options> {}
+
+// DefaultOptions declares the default options for a Logger
+export const DefaultOptions: Options = {
+    level: LogLevel.INFO,
+    serialize: JSONSerializer,
+    stamps: [RFC3339Stamper],
+    write: StdOutWriter,
+}
+
 // Logger class allows consumer to instantiate a logger with any given level / writer
 export class Logger {
     // context holds any extra data to include in all logs from this logger
     private context: any
-
-    constructor(
-        private readonly _level: LogLevel = LogLevel.INFO,
-        private readonly write: Writer = StdOutWriter,
-        private readonly serialize: Serializer = JSONSerializer,
-        private readonly stamps: Stamper[] = [RFC3339Stamper],
-    ) {}
+    public readonly options: Options
+    constructor(params: Parameters) {
+        this.options = { ...DefaultOptions, ...params }
+    }
```

Comment:
this could optionally be a non-breaking change by checking the parameter type 🤔
```suggestion
    constructor(
        params: LogLevel | Options = LogLevel.INFO,
        write?: Writer,
        serialize?: Serializer,
        stamps?: Stamper[],
    ) {
        if ( params as LogLevel in LogLevel ) {
            this.options = { ...DefaultOptions }
            this.options.level = params as LogLevel
            this.options.write = write ? write : DefaultOptions.write
            this.options.serialize = serialize ? serialize : DefaultOptions.serialize
            this.options.stamps = stamps ? stamps : DefaultOptions.stamps
        } else {
            this.options = { ...DefaultOptions, ...params as Options }
        }
    }
```